### PR TITLE
Active payment

### PR DESCRIPTION
**Pull request by demands14**
## Related Tickets
- [#51773](https://edu-redmine.sun-asterisk.vn/issues/51773)
- [#51774](https://edu-redmine.sun-asterisk.vn/issues/51774)
- [#51771](https://edu-redmine.sun-asterisk.vn/issues/51771)
- [#51780](https://edu-redmine.sun-asterisk.vn/issues/51780)
## HOW
- create payments, order_history, payment_activations  controller, view

## Evidence (Screenshot or Video)
image
image

### CODE REVIEW

**Comment by dongpx1491** [comment]
Ready

**Comment by dongpx1491** [comment]
Ready

**Comment by demands14** [comment]
Ready

File: app/controllers/order_historys_controller.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class OrderHistorysController < ApplicationController
+  def index
+    @pagy, @payments = pagy current_user.payments.show_active.latest
+  end
+
+  def show
+    @payment = Payment.find_by id: params[:id]
```
Comment (huongnt-2545) [inline app/controllers/order_historys_controller.rb:12]: thêm xử lý trg hợp nil

Comment (huongnt-2545) [inline app/controllers/order_historys_controller.rb:12]: thêm xử lý trường hợp nil

File: app/views/tickets/_ticket.html.erb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    <ul>
+      <li class="xsize"><%= t ".price" %> <%= ticket.seat.price %></li>
+      <li class="xsize"><%= t ".seat" %> <%= "#{ticket.seat.type_seat} #{ticket.seat.id}" %></li>
+      <% unless ticket.payment.active? %>
+        <li class="xsize delete"><%= link_to t(".delete"), ticket, method: :delete,
+          remote: true, data: {confirm: t(".confirm")}, class: "delete_a" %></li>
+      <% end %>
+    </ul>
+    <h3 class="title_h3"><%= ticket.show.movie.title %></h3>
+    <span class="xsize_100px"><i class="fa-solid fa-calendar-days"></i> <%= ticket.show.date %></span>
```
Comment (huongnt-2545) [inline app/views/tickets/_ticket.html.erb]: `ticket.show.date` -> dùng delegate

File: app/views/tickets/_ticket.html.erb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    <span class="span_cinema"><i class="fa-solid fa-clapperboard"></i>22 <%= ticket.show.room.name %></span>
```
Comment (huongnt-2545) [inline app/views/tickets/_ticket.html.erb]: `22` là cái gì thế e

File: config/routes.rb [hunk 4]
```diff
@@ -1,4 +1,6 @@
 Rails.application.routes.draw do
+  get 'order_historys/index'
+  get 'order_historys/show'
```
Comment (huongnt-2545) [inline config/routes.rb]: xóa đi em

File: spec/fixtures/payment_mailer/confirm_payment [hunk 5]
```diff
@@ -0,0 +1,3 @@
+PaymentMailer#confirm_payment
```
Comment (huongnt-2545) [inline spec/fixtures/payment_mailer/confirm_payment]: xóa file

File: spec/fixtures/payment_mailer/payment_activation [hunk 6]
```diff
@@ -0,0 +1,3 @@
+PaymentMailer#payment_activation
```
Comment (huongnt-2545) [inline spec/fixtures/payment_mailer/payment_activation]: xóa file

File: spec/helpers/payment_activations_helper_spec.rb [hunk 7]
```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
```
Comment (huongnt-2545) [inline spec/helpers/payment_activations_helper_spec.rb]: xóa file

File: spec/mailers/payment_mailer_spec.rb [hunk 8]
```diff
@@ -0,0 +1,32 @@
+require "rails_helper"
```
Comment (huongnt-2545) [inline spec/mailers/payment_mailer_spec.rb]: xóa file

**Comment by demands14** [comment]
Ready

**Review by huongnt-2545 — COMMENTED** [review]
cmt
còn comment chưa fix nhé

**Comment by demands14** [comment]
Ready

**Review by huongnt-2545 — APPROVED** [review]
merged

**Pull request merged**